### PR TITLE
MM-41042: Allow trusted plugin routes

### PR DESCRIPTION
**Pull request by agarciamontoro**
#### Summary
This PR adds a function to allow trusted plugin routes. This is designed to be easily expanded in the future for other routes but, for now, it only allows the `/api/v1/export` endpoint from the channel export plugin.

By allowing that specific endpoint here, we fix [the bug](https://mattermost.atlassian.net/browse/MM-41042) that prevented the channel log from being downloaded.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-41042

#### Checklist

- [X] Added or updated unit tests (required for all new features)
- [ ] Has UI changes
- [X] read and understood our [Contributing Guidelines](https://github.com/mattermost/desktop/blob/master/CONTRIBUTING.md)
- [X] completed [Mattermost Contributor Agreement](http://www.mattermost.org/mattermost-contributor-agreement/)
- [X] executed `npm run lint:js` for proper code formatting

#### Device Information
This PR was tested on: laptop, Linux

#### Screenshots
https://user-images.githubusercontent.com/3924815/150202926-519882d9-fb34-42f9-a0c1-013e39e733bc.mp4

#### Release Note
```release-note
Fixed a bug that prevented the channel log from being downloaded from Playbooks.
```

### CODE REVIEW

**Comment by agarciamontoro** [comment]
Not sure if I added the right reviewers here. Feel free to unassign yourself if needed!

File: src/common/utils/url.ts
Comment (devinbinnie) [reply src/common/utils/url.ts:222]: I think it would make sense to just check for `isPluginRoute` and then checking to see if it's the specific path we have below. I would add those paths into an array of some kind and check if it's one of those special cases.

**Review by devinbinnie — CHANGES_REQUESTED** [review]

File: src/common/utils/url.ts
Comment (agarciamontoro) [reply src/common/utils/url.ts:222]: The thing is that we still need to get the server info, parse the URL and check the prefix in order to clean the pathname to make sure we check against a well-defined path. So it would end up like this:

```ts
    if (!isPluginUrl(serverUrl, inputUrl)) {
        return false;
    }

    // We still need to get the server and parse the URL, which isPluginUrl already did
    const server = getServerInfo(serverUrl);
    const parsedUrl = parseURL(inputUrl);

    // We also need to check that they're not undefined for typescript
    // to not complaint, even if isPluginUrl already checked
    if (!server || !parsedUrl) {
        return false;
    }

    const formattedPathName = parsedUrl.pathname.toLowerCase();
    
    // These checks were also done already
    let cleanPathname;
    if (formattedPathName.startsWith(`${server.subpath}plugins/`)) {
        cleanPathname = formattedPathName.replace(`${server.subpath}plugins/`, '');
    } else 
        // this branch now relies on the actual implementation of isPluginUrl.
        // we could alternatively still check for startsWith('/plugins') in an else if branch
        // and return false in the final else
        formattedPathName.replace('/plugins/', '');
    }

    const allowedRoutes = [
        'com.mattermost.plugin-channel-export/api/v1/export',
    ];

    return allowedRoutes.includes(cleanPathname);
```

This is more or less the same we had before, but now we have code running twice (check the comments above) and rely on the inner implementation of `isPluginUrl`.

Maybe there is some other utility function already written for making this easier? If not, I rather keep the actual implementation.

Thoughts?

Comment (devinbinnie) [reply src/common/utils/url.ts:222]: If the route is always going to be of the form `/plugins/com.mattermost.plugin-channel-export/api/v1/export`, maybe instead we try to keep it simple for now and just write another helper that just calls `isUrlType` instead of first checking if it's a plugin url and THEN checking if its the exception.

```suggestion
function isChannelExportUrl(serverUrl: URL | string, inputURL: URL | string) {
    return isUrlType('plugins/com.mattermost.plugin-channel-export/api/v1/export', serverUrl, inputURL);
}
```

Comment (agarciamontoro) [reply src/common/utils/url.ts:222]: Yeah, that makes more sense. Even my original proposal was premature generalization :) Let me test that and push it.
 Thanks for the discussion!

**Review by devinbinnie — APPROVED** [review]
Thanks @agarciamontoro :)

**Comment by agarciamontoro** [comment]
@Willyfrog, do you think you can review this today or tomorrow to get this in 5.0.3?

**Review by Willyfrog — APPROVED** [review]

**Comment by agarciamontoro** [comment]
Hm, I have an unrelated? failing test :thinking: Is that a known flaky test?

**Comment by devinbinnie** [comment]
> Hm, I have an unrelated? failing test 🤔 Is that a known flaky test?

Yeah it's a flaky test, might have to look at that one again. Re-ran the checks for ya :)

**Comment by agarciamontoro** [comment]
Thanks! I tried to do that myself, but it seems I have no permissions to click the right button.

**Comment by agarciamontoro** [comment]
Not sure what the PR process for the desktop app is. Do we need QA here? I'd say we don't, but I wanted to ask before clicking the Merge button :)

**Comment by devinbinnie** [comment]
Nope, good to merge!

**Pull request merged**
